pkg/uio/reader.go

```diff
@@ -46,3 +47,18 @@ func ReaderAtEqual(r1, r2 io.ReaderAt) bool {
 	}
 	return bytes.Equal(c, d) && reflect.DeepEqual(r1err, r2err)
 }
+
+// ReadIntoFile reads all from io.Reader into the file at given path.
+//
+// If the file at given path does not exist, a new file will be created.
+// If the file exists at the given path, but not empty, it will be truncated.
+func ReadIntoFile(r io.Reader, p string) error {
+	f, err := os.OpenFile(p, os.O_CREATE|os.O_RDWR|os.O_TRUNC, 0644)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
```

Comment:
Check the error returned by Close.

Reply:
Which kind of error scenario worth attention ?

my understanding is that f.Close() renders the file description un-available for IO, and return an error when it is already called. I weakly recall that is POSIX style, I did not deal with others. I could certainly refresh my understanding here.

Reply:
Close can have an error. 
Here is the scenario. 
you call write lots of times. All that data is buffered. You call close. Close means, besides the expected "yeah it's fine", "report back on whether all the data made it to disc". CA 1988, when vi did not check this, people were sad to do a :wq and then find .. the file was empty. 

Back in the day, close only returned EBADFD. Now it's got stuff like ENOSPC, EIO, etc, and it's best to check.

Reply:
Done.

That tempts me to do a `return f.Sync()` in the end, but that adds penalty in time to do disk IOs, and lead to reopening of the file descriptor, says , `Dealing with error returns from close()`, from https://man7.org/linux/man-pages/man2/close.2.html

Reply:
Calling Close twice in Go is totally fine. You do not need to worry about FD reuse.

The second Close() on an *os.File will return an error (and the second call will be the defer'd one, so that error is ignored).